packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -2052,6 +2053,65 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     _maybeStartFadeoutTimer();
   }
 
+  // Returns the delta that should result from applying [event] with axis and
+  // direction taken into account.
+  double _pointerSignalEventDelta(PointerScrollEvent event) {
+    assert(_cachedController != null);
+    double delta = _cachedController!.position.axis == Axis.horizontal
+        ? event.scrollDelta.dx
+        : event.scrollDelta.dy;
+
+    if (axisDirectionIsReversed(_cachedController!.position.axisDirection)) {
+      delta *= -1;
+    }
+    return delta;
+  }
+
+  // Returns the offset that should result from applying [event] to the current
+  // position, taking min/max scroll extent into account.
+  double _targetScrollOffsetForPointerScroll(double delta) {
+    assert(_cachedController != null);
+    return math.min(
+      math.max(_cachedController!.position.pixels + delta, _cachedController!.position.minScrollExtent),
+      _cachedController!.position.maxScrollExtent,
+    );
+  }
+
+  void _handlePointerScroll(PointerEvent event) {
+    assert(event is PointerScrollEvent);
+    _cachedController = _effectiveScrollController;
+    final double delta = _pointerSignalEventDelta(event as PointerScrollEvent);
+    final double targetScrollOffset = _targetScrollOffsetForPointerScroll(delta);
+    if (delta != 0.0 && targetScrollOffset != _cachedController!.position.pixels) {
+      _cachedController!.position.pointerScroll(delta);
+    }
+    _cachedController = null;
+  }
+
+  void _receivedPointerSignal(PointerSignalEvent event) {
+    _cachedController = _effectiveScrollController;
+    // Only try to scroll if the bar absorb the hit test.
+    if ((scrollbarPainter.hitTest(event.localPosition) ?? false) &&
+        _cachedController != null &&
+        _cachedController!.hasClients &&
+        !_thumbDragging) {
```

Comment:
`_thumbDragging` is intend to do not break
https://github.com/Piinks/flutter/blob/f574242ca5df1bf2ad1bd0000f75e6a35ed25a92/packages/flutter/test/material/scrollbar_test.dart#L1371-L1374

But I test on the window native platform, it can pointer scroll during drag. WDYT @Piinks

Reply:
I need to look at this more, but it sounds like it might be different behavior on different platforms?

Reply:
Probably.

Reply:
Sorry for the delay, I was out sick.

I think that test was added to ensure we did not regress on a crash wayy back when CupertinoScrollbar implemented its own scrollbar drag gesture. The way it did this was to create a drag event on the scrollable, and so it would crash if more than one drag was happening simultaneously. Since the scrollbar gestures were refactored and they do not create a drag event anymore, it is probably safe to remove that test and allow thumb dragging. Since this is a new feature, we can iterate on it as we compare it to other platforms.

Reply:
I am planning to allow this behavior only on the web. WDYT? @Piinks

Reply:
Thank you for researching this so fully!

> I am planning to allow this behavior only on the web

I think that makes sense. 👍

Reply:
Will continue to update after #112434  is checked in, they may have conflicts.